src/realm/object-store/sync/app_utils.cpp

```diff
@@ -22,18 +22,38 @@
 
 #include <external/json/json.hpp>
 
+#include <algorithm>
+
 namespace realm {
 namespace app {
 
+static bool string_iequals(const std::string& a, const std::string& b)
+{
+    return std::equal(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+        return tolower(a) == tolower(b);
+    });
+}
+
+util::Optional<std::pair<const std::string, std::string>>
+AppUtils::find_header(const std::string& key_name, const std::map<std::string, std::string>& search_map)
+{
+    for (auto current : search_map) {
+        if (string_iequals(key_name, current.first)) {
```

Comment:
```suggestion
        if (keyname.size() == current.first.size() && strcasecmp(key_name.c_str(), current.first.c_str()) == 0) {
```

This will be a lot faster than going byte-by-byte and doing `tolower` on both sides.